lib/tapioca/gem/pipeline.rb

```diff
@@ -240,10 +240,7 @@ def compile_object(name, value)
 
         klass = class_of(value)
 
-        klass_name = if klass == ObjectSpace::WeakMap
-          # WeakMap is an implicit generic with one type variable
-          "ObjectSpace::WeakMap[T.untyped]"
-        elsif T::Generic === klass
+        klass_name = if T::Generic === klass
```

Comment:
Wouldn't this may create an error depending on the version of Sorbet you're using?

I think we should register this as a feature in [SorbetHelper](https://github.com/Shopify/tapioca/blob/main/lib/tapioca/helpers/sorbet_helper.rb) and generate the correct type depending on `sorbet_supports?`